src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -318,6 +318,22 @@ private void Populate(ref LocalState state, int start)
             }
         }
 
+        private static TypeSymbolWithAnnotations GetAdjustedType(TypeSymbolWithAnnotations initialType)
+        {
+            // If the initial type was an unconstrained type parameter, we want to treat it as nullable, overriding
+            // the annotated state.
```

Comment:
Why not return `IsNullable == true` from the original `TypeSymbolWithAnnotations`?

Reply:
Because the original TWSA is from initial binding, and `IsNullable` does not return true.

---
In reply to: [206707381](https://github.com/dotnet/roslyn/pull/28956#discussion_r206707381) [](ancestors = 206707381)

Reply:
Why not set `treatUnconstrainedGenericsAsNullable: true` in initial binding?

---
In reply to: [206722195](https://github.com/dotnet/roslyn/pull/28956#discussion_r206722195) [](ancestors = 206722195,206707381)

Reply:
Talked with @cston offline. We'll defer doing this for a later PR, I'll add a prototype comment to track.

---
In reply to: [206750233](https://github.com/dotnet/roslyn/pull/28956#discussion_r206750233) [](ancestors = 206750233,206722195,206707381)